File: models/entregaMaterial.js
module.exports = (sequelize, DataTypes) => {
  const EntregaMaterial = sequelize.define('EntregaMaterial', {
    data_entrega: DataTypes.DATE,
    qtd_pontos_gerados: DataTypes.BIGINT,
    valor_total_gerado: DataTypes.DECIMAL(10, 2),
  }, {
    tableName: 'tb_entrega_material',
    timestamps: false,
  });

  EntregaMaterial.associate = (models) => {
    EntregaMaterial.belongsTo(models.Usuario, { foreignKey: 'id_user' });
    EntregaMaterial.belongsTo(models.Ecoponto, { foreignKey: 'id_ecoponto' });
    EntregaMaterial.belongsTo(models.Empresa, { foreignKey: 'id_empresa' });
    EntregaMaterial.hasMany(models.EntregaTipoLixo, { foreignKey: 'id_entrega_material' });
  };

  return EntregaMaterial;
};

File: src/main/java/org/acme/models/EntregaMaterial.java
package org.acme.models;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "tb_entrega_material")
public class EntregaMaterial {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "data_entrega")
    private LocalDateTime dataEntrega;
    @Column(name = "qtd_pontos_gerados")
    private Long qtdPontosGerados;
    @Column(name = "id_empresa")
    private Long idEmpresa;
    @Column(name = "valor_total_gerado")
    private Long valorTotalGerado;
    @Column(name = "deferido")
    private Boolean deferido;
    @ManyToOne
    @JoinColumn(name = "id_user")
    private Usuario usuario;


    @ManyToOne
    @JoinColumn(name = "id_ecoponto")
    private Ecoponto ecoponto;
    @ManyToOne
    @JoinColumn(name = "id_status")
    private StatusSolicitacao status;

}
